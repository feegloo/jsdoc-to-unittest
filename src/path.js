export default ([base, ...parts], isFunction) => {
  let path = base;
  parts.forEach((part) => {
    if (Number.isNaN(+part) || typeof part !== 'string') {
      path += `.${part}`;
    } else {
      path += `[${part}]`;
    }
  });
  if (isFunction) {
    path += '()';
  }

  return path;
};
