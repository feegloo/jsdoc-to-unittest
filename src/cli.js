import fs from 'fs';
import path from 'path';
import util from 'util';
import { argv } from 'yargs';
import print from './creator';

const readFileAsync = util.promisify(fs.readFile);

const basePath = path.resolve(__dirname, '..');

const fields = [
  'inline'
];

async function getConfig(input, output) {
  let configPath = '';
  const config = { // default values
    export: true,
  };
  if (argv.config) {
    configPath = path.resolve(basePath, argv.config, '..');
    Object.assign(config, JSON.parse(
      await readFileAsync(path.resolve(basePath, argv.config), 'utf-8')
    ));
  }

  fields.forEach((field) => {
    if (field in argv) {
      config[field] = argv[field];
    }
  });

  Object.assign(config, {
    input,
    output,
  });

  if (argv.header) {
    config.header = await readFileAsync(path.resolve(basePath, config.header), 'utf-8');
  } else if (config.header) {
    config.header = await readFileAsync(path.resolve(configPath, config.header), 'utf-8');
  }
  if (argv.footer) {
    config.footer = await readFileAsync(path.resolve(basePath, config.footer), 'utf-8');
  } else if (config.footer) {
    config.footer = await readFileAsync(path.resolve(configPath, config.footer), 'utf-8');
  }

  return config;
}

export default async (input = argv._[0], output = argv._[1]) => {
  const config = await getConfig(input, output);
  return print({
    ...config,
    get stdout() {
      return output === undefined || String(output).trim().length === 0;
    },
  });
};
