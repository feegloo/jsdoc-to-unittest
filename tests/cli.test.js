import path from 'path';
import cli from 'src/cli';

describe('#cli', () => {
  test('creates test', async () => {
    expect(await cli(path.resolve(__dirname, './fixtures/contains.js'), '')).toMatchSnapshot();
  });
});
