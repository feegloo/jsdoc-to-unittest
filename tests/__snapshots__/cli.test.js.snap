// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#cli array-complement.js matches snapshot 1`] = `
"export function arrayComplement(array) {
  const indexes = Array.prototype.slice.call(arguments, 1);
  return easy.utils.filter(array, (value, arrIndex) => !easy.utils.inArray(indexes, arrIndex));
}
export default { arrayComplement };

/* global mock */
// Valid tests: 100.00%

describe('arrayComplement', () => {
  test('example 1', () => {
    expect(
      evaluate(
        () => {
          const arr = [10, 5, 3];
          frosmo.site.utils.arrayComplement(arr, 1);
        },
        func => eval(func),
      ),
    ).toEqual([10, 3]);
  });

  test('example 2', () => {
    expect(
      evaluate(
        () => {
          const arr = ['a', 'b', 'c'];
          frosmo.site.utils.arrayComplement(arr, 0, 2);
        },
        func => eval(func),
      ),
    ).toEqual(['b']);
  });
});
"
`;

exports[`#cli creates async test 1`] = `
"export function getFromURL(str) {
  return fetch(str);
}
export default { getFromURL };

/* global mock */
// Valid tests: 100.00%

describe('getFromURL', () => {
  test('example 1', async () => {
    expect(
      await mock.async(
        () =>
          getFromURL('https://mail.google.com/mail/').then(response => response.slice(5).trim()),
        { \\"fetch('string')\\": 'hello world  ' },
        func => eval(func),
      ),
    ).toBe('world');
  });
});
"
`;

exports[`#cli creates test 1`] = `
"export function contains(str1, str2) {
  return str1 && str1.indexOf(str2) !== -1;
}
export default { contains };

/* global mock */
// Valid tests: 100.00%

describe('contains', () => {
  test('example 1', () => {
    expect(contains('foo', 'fooo')).toBe(false);
  });

  test('example 2', () => {
    expect(contains('hey', 'ey')).toBe(true);
  });
});
"
`;
